### ISSUE TITLE
Problems to install mod_gllog in Apache

### ISSUE THREAD

**Issue by rdelatorrea07** [mikkolehtisalo/mod_gllog#2]
Hello,

I followed your instructions to install mod_gllog and I found the following problems;
FIRST ERROR:
The problem was " mod_gllog.c : 258: error: ' AP_DECLARE_MODULE ' function initializes a variable" (error in the source code ) .

I modified the source code and I could compile.

SECOND ERROR:
When configuring the module in Apache, I get the error:

Starting httpd : httpd : Syntax error in line 204 of /etc/httpd/conf/httpd.conf : API module structure ' gllog_module ' in the file
/usr/lib64/httpd/modules/mod_gllog.so is illegible - 41503232 expects the firm, but saw 00000000 - maybe this is not an Apache module DSO , or was compiled for a different version of Apache ?

Can you help me?
Thank you very much for your input and attention
Regards!

**Comment by mikkolehtisalo** [comment]
What is the exact distro you are trying to install on ?

**Comment by rdelatorrea07** [comment]
My distro is CentOS 6.5

**Comment by rdelatorrea07** [comment]
Hi,
I enclose you more information;
**# httpd -V**
Server version: Apache/2.4.18 (Unix)
Server built:   Mar  2 2016 11:52:35
Server's Module Magic Number: 20120211:52
Server loaded:  APR 1.5.2, APR-UTIL 1.5.4
Compiled using: APR 1.5.2, APR-UTIL 1.5.4
Architecture:   64-bit
Server MPM:     event
  threaded:     yes (fixed thread count)
    forked:     yes (variable process count)
Server compiled with....
 -D APR_HAS_SENDFILE
 -D APR_HAS_MMAP
 -D APR_HAVE_IPV6 (IPv4-mapped addresses enabled)
 -D APR_USE_SYSVSEM_SERIALIZE
 -D APR_USE_PTHREAD_SERIALIZE
 -D SINGLE_LISTEN_UNSERIALIZED_ACCEPT
 -D APR_HAS_OTHER_CHILD
 -D AP_HAVE_RELIABLE_PIPED_LOGS
 -D DYNAMIC_MODULE_LIMIT=256
 -D HTTPD_ROOT="/usr/local/httpd-2.4.18"
 -D SUEXEC_BIN="/usr/local/httpd-2.4.18/bin/suexec"
 -D DEFAULT_PIDLOG="logs/httpd.pid"
 -D DEFAULT_SCOREBOARD="logs/apache_runtime_status"
 -D DEFAULT_ERRORLOG="logs/error_log"
 -D AP_TYPES_CONFIG_FILE="conf/mime.types"
 -D SERVER_CONFIG_FILE="conf/httpd.conf"
**#httpd -l**
Compiled in modules:
  core.c
  mod_so.c
  http_core.c
  event.c

Thank you for your attention
Regards!

**Comment by mikkolehtisalo** [comment]
Where is that httpd installed from? Centos 6.x comes with 2.2. Epel doesn't have 2.4 either for 6.x.

**Comment by rdelatorrea07** [comment]
Hello,
I had Apache 2.2.29 , but I could not install gllog and I decided to install the latest stable version (2.4.18) of Apache but does not solve the problem.
I think that the problem is not the version of Apache

Regards!

**Comment by mikkolehtisalo** [comment]
Some of the software versions on Centos 6.x are a bit old.. In general you really should be using 7.x already. Anyways, I installed a minimal VM and this is how I got the module to compile and work:

```
# Common packages you need
yum groupinstall 'Development Tools'
yum install httpd httpd-devel openssl-devel

# Compile and install apr 1.5.2
tar xvzf apr-1.5.2.tar.gz 
cd apr-1.5.2
./configure --prefix=/opt/apr-1.5.2
make
make install

# Preload the apr 1.5.2, this is Centos 6.x
echo "/opt/apr-1.5.2/lib" >> /etc/ld.so.conf.d/apr152.conf
ldconfig

# Modify the last part of mod_gllog.c to be Apache 2.2 compatible
module AP_MODULE_DECLARE_DATA gllog_module =
{
    STANDARD20_MODULE_STUFF,
    NULL,
    NULL,
    GlLogCreateServerConfig,
    NULL,
    GlLogCmds,
    GlLogRegisterHooks
};

# It should now compile
apxs -I/opt/apr-1.5.2/include/apr-1 -L/opt/apr-1.5.2/lib -i -c mod_gllog.c

# Test it with something like


NameVirtualHost *:80
LoadFile /usr/lib64/libssl.so
LoadModule gllog_module /usr/lib64/httpd/modules/mod_gllog.so

<VirtualHost *:80>
    ServerAdmin webmaster@dummy-host.example.com
    DocumentRoot /www/docs/dummy-host.example.com
    ServerName dummy-host.example.com
    ErrorLog logs/dummy-host.example.com-error_log
    CustomLog logs/dummy-host.example.com-access_log common
    GlLog On
    GlLogKey "ae9a6138c6285163f1820e68cffda6191e0717e064917ee73642c819ca8b3e39f2e60d847f2a9f585ac53479f45365cb40cbaa7d1b102b8e5181704d695dedca"
    GlLogSignature On
    CustomLog ${APACHE_LOG_DIR}/access.log gl_combined
</VirtualHost>

```

**Comment by rdelatorrea07** [comment]
hi

Great job, I can intall mod_gllog.

Now I have a new problem, the results look like the following:

_192.168.100.4 - - [03/Mar/2016:13:13:56 +0100] "GET /a/connection/available HTTP/1.1" 200 458 "https://192.168.100.179/login" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko" "-" "-" "invalid"
192.168.100.4 - - [03/Mar/2016:13:18:35 +0100] "POST /a/metrics/205/1qenj1g1/xhr?t=1457007401883 HTTP/1.1" 200 1129 "https://192.168.100.179/gettingstarted" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko" "-" "-" "invalid"
192.168.100.4 - - [03/Mar/2016:13:13:54 +0100] "GET /assets/fonts/OpenSans-Light-webfont.eot? HTTP/1.1" 304 197 "https://192.168.100.179/login" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko" "-" "-" "invalid"
192.168.100.4 - - [03/Mar/2016:13:13:56 +0100] "GET /a/connection/available HTTP/1.1" 200 458 "https://192.168.100.179/login" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko" "-" "-" "invalid"_

I can not see the name of the authenticated neither the session id. This fields are blank ("-" "-").

Please, Can you help me?

Thanks you very much.

Regards.
